models/DataObject/Concrete.php

```diff
@@ -742,7 +742,10 @@ public function __wakeup(): void
             // We're reloading version data, there might be fields that now implement the ObjectAwareFieldInterface but
             // aren't included in the $this->__objectAwareFields array - for example versions created in Pimcore <= 10.x
             // containing LocalizedFields. Verify all fields in this object.
-            foreach (get_object_vars($this) as $propertyValue) {
+            foreach (get_object_vars($this) as $propertyKey => $propertyValue) {
+                if(str_ends_with($propertyKey, 'o_published')) {
+                    $this->published = (bool) $propertyValue;
+                }
```

Comment:
```suggestion
                if(str_starts_with($propertyKey, 'o_')) {
                    $propertyKeyNew = preg_replace('/^o_/', '', $propertyKey);
                    if(property_exists($this, $propertyKeyNew)) {
                        $this->$propertyKeyNew = (bool) $propertyValue;
                    }
                }
```

I think we need to make this more generic, since other fields which were renamed from `o_` could be affected as well, right?

Reply:
@mhgtlc any updates on this? 🏓 Thanks!

Reply:
@brusch I don't think this is a good ideea because some propertes are protected or private, and in this case it starts with some special characters, not with `o_`

Reply:
@mhgtlc Hmmm, but `published` is now also `protected`. 
I'm just thinking about properties like: 
```
key
index
creationDate
modificationDate
...
```

They all were `public` before and prefixed with `o_` 🤔